### PR TITLE
fix: [CDS-42730]: UX improvements for linking deployment template

### PR DESCRIPTION
**Pull request by samarth-sharma31**
#### Screenshots

NA

Screenshot 2022-10-07 at 12 02 50 PM

https://user-images.githubusercontent.com/96409598/194521917-850de9b6-dd81-41f0-8ea9-90bd3b320f9c.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 6 Issues 
* Code issue 6 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11736)

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

File: src/framework/Templates/TemplateSelectorContext/useTemplateSelector.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export interface GetTemplateProps {
   linkedTemplate?: {
     identifiers: string[]
     checkboxLabel?: string
+    showCheckbox?: boolean
```
Comment (yc147) [inline src/framework/Templates/TemplateSelectorContext/useTemplateSelector.ts:26]: Ideally there is no need to add this boolean `showCheckbox`.  Just hide the checkbox `isEmpty(linkedTemplate)` is `true`.

Comment (samarth-sharma31) [reply src/framework/Templates/TemplateSelectorContext/useTemplateSelector.ts:26]: This boolean has been added to support the new use case for deployment template --> the checkbox is not to be shown even if linkedTemplate is present -> that's why isEmpty(linkedTemplate) won't be enough now --> visibility of checkbox also needs to be controlled.

Comment (yc147) [reply src/framework/Templates/TemplateSelectorContext/useTemplateSelector.ts:26]: Okay 👍

**Pull request merged**